main.cpp

```diff
@@ -49,77 +49,105 @@ Create a branch named Part1
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if ( inputValue != nullptr && that != nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "\nU's var1 value: " << that->var1 << std::endl;
+            that->var1 = *inputValue; 
+            std::cout << "\nU's var1 updated value: " << that->var1 << std::endl;
+            std::cout << "Reducing magnitude.. \n";
+            while( std::abs (static_cast <long> (that->var2 - that->var1)) > 0.001f )
+                {
+                    /*
+                    write something that makes the distance between that->name2 and that->name1 get smaller
+                     */          
+                    that->var2 += 0.5f;
+                    std::cout << that->var2 << ", ";
+                }
+        std::cout << "\nU's var2 updated value: " << that->var2 << std::endl;
+        return (that->var2 * that->var1);
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        return *inputValue; 
     }
 };
         
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+
+    T var1( 0.1f , "a" );                                             
+    T var2( 0.5f , "b");  
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Operation f;                                      
+    T* smaller = f.compare( &var1, &var2 );  
+    if (smaller != nullptr) std::cout << "\n\nthe smaller one is << " << smaller->name << std::endl; 
+    else std::cout << "Something is wrong with the compare function..." << std::endl; 
     
-    U <#name3#>;
-    float updatedValue = 5.f;
-    std::cout << "[static func] <#name3#>'s multiplied values: " << <#structname2#>::<#staticFunctionA#>( , ) << std::endl;                  //11
+    U user3 ( -60.0f  );
+    float updatedValue = 30.0f;
+    std::cout << StaticTransform::product( &user3 , &updatedValue ) << " is the result of StaticTransform func user3's multiplied values." << std::endl;   
     
-    U <#name4#>;
-    std::cout << "[member func] <#name4#>'s multiplied values: " << <#name4#>.<#memberFunction#>( &updatedValue ) << std::endl;
+    U user4 ( -60.0f  );
+    std::cout << user4.product( &updatedValue ) << " is the result of Non-Static member func user4's multiplied values. " << std::endl;
```

Comment:
Please leave the `std::cout` statements and variables as they appeared on the left. Just replace the placeholders as instructed.

Don't write more code than you need to.

Reply:
`U user4 ( -60.0f  );
std::cout << "Product function user4's multiplied values: " << user4.product( &updatedValue ) << std::endl;`